BioCro/DARPA/biocro_regional_darpa_files/subset_weather.Rmd

```diff
@@ -0,0 +1,148 @@
+
+ * Latitudinal range: 10.125 - 83.875 
+ * Longitudinal range: -169.875 to -50.125
+
+The entire global file is the CRUNCEP `all.nc`, which is ~1.3TB and in the same Globus collection [here](https://app.globus.org/file-manager?origin_id=cbae3a96-d081-4951-bd1e-a8fda974cefa&origin_path=%2Fmet%2Fcruncep%2F). We will need more data space on the HPC to subset this because the upper limit is currently 1TB. 
+
+## Subsetting process
+
+### 1. Copy entire data file onto HPC
+
+First increase data space by requesting an xdisk allocation on the HPC. Go to [the HPC web interface](https://ood.hpc.arizona.edu/pun/sys/dashboard), and open up the terminal by clicking on Files -> Home Directory -> Open in terminal -> Ocelote. Use `xdisk -c create -m 1000` to get 1TB of space.  
+
+Open up the `all.nc` file on Globus using the link above. Then open up the HPC collection, which is called `arizona#sdmz-dtn`. Navigate to the new terrabyte of space in this collection at `/xdisk/username/`. Use the "Start" button to initiate transfer, which will take about an hour. Also transfer `champaign.nc` if desired. 
+
+### 2. Determine desired dimensions
+
+Use the following plotting functionality in R to figure out which latitudes and longitudes to clip the weather data to. 
+
+```{r message=FALSE, warning=FALSE}
+library(ncdf4)
+library(dplyr)
+library(maps)
+library(ggplot2)
+```
+
+The first example shows the four corners of a long latitudinal slice that includes area in both China and Russia. Change out the `lat` and `lon` values in the data frame and the filtered region in the `background_map` as desired. 
+
+```{r}
+chiruss_corners = expand.grid(lat = c(44.9, 55.1), 
+                             lon = c(123.9, 127.1))
+background_map <- map_data("world") %>% 
+  filter(region == "China" | region == "Russia")
+ggplot() +
+  geom_polygon(data = background_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") +
+  geom_point(data = chiruss_corners, aes(x = lon, y = lat), color = "red")
+```
+
+This plot shows the limits of the NARR `all.nc` file. 
+
+```{r}
+narr_all_corners = expand.grid(lat = c(10.125, 83.875), 
+                             lon = c(-169.875, -50.125))
+background_map <- map_data("world")
+ggplot() +
+  geom_polygon(data = background_map, aes(x = long, y = lat, group = group), fill = "white", color = "black") +
+  geom_point(data = narr_all_corners, aes(x = lon, y = lat), color = "red")
+```
+
+### 3. Use nco to subset file
+
+Navigate in the HPC web interface to `/xdisk/username/`. The .nc files should be there now. You can see the subsetting command by looking at the first line under history after running this: 
```

Comment:
Perhaps add a grep command to focus on history. Perhaps something like `ncdump -h campaign.nc | grep ':history'`?